compiler/qsc_rca/tests/common/mod.rs

```diff
@@ -0,0 +1,160 @@
+        let package_compute_properties = self.compute_properties.get_mut(package_id);
+        package_compute_properties.clear();
+        let analyzer = Analyzer::init_with_compute_properties(
+            &self.fir_store,
+            self.compute_properties.clone(),
+        );
+        self.compute_properties = analyzer.analyze_package(package_id);
+    }
+}
+
+impl Default for CompilationContext {
+    fn default() -> Self {
+        let compiler = Compiler::new(
+            true,
+            SourceMap::default(),
+            PackageType::Lib,
+            RuntimeCapabilityFlags::all(),
+            LanguageFeatures::default(),
+        )
+        .expect("should be able to create a new compiler");
+        let mut lowerer = Lowerer::new();
+        let fir_store = lower_hir_package_store(&mut lowerer, compiler.package_store());
+        let analyzer = Analyzer::init(&fir_store);
+        let compute_properties = analyzer.analyze_all();
+        Self {
+            compiler,
+            fir_store,
+            lowerer,
+            compute_properties,
+        }
+    }
+}
+
+pub trait PackageStoreSearch {
+    fn find_callable_id_by_name(&self, name: &str) -> Option<StoreItemId>;
+}
+
+impl PackageStoreSearch for PackageStore {
+    fn find_callable_id_by_name(&self, name: &str) -> Option<StoreItemId> {
+        for (package_id, package) in self.iter() {
+            if let Some(item_id) = package.find_callable_id_by_name(name) {
+                return Some((package_id, item_id).into());
+            }
+        }
+
+        None
+    }
+}
+
+pub trait PackageSearch {
+    fn find_callable_id_by_name(&self, name: &str) -> Option<LocalItemId>;
+}
+
+impl PackageSearch for Package {
+    fn find_callable_id_by_name(&self, name: &str) -> Option<LocalItemId> {
+        for (item_id, item) in &self.items {
+            if let ItemKind::Callable(callable_decl) = &item.kind {
+                if callable_decl.name.name.as_ref() == name {
+                    return Some(item_id);
+                }
+            }
+        }
+
+        None
+    }
+}
+
+pub fn check_callable_compute_properties(
+    fir_package_store: &impl PackageStoreSearch,
+    package_store_compute_properties: &PackageStoreComputeProperties,
+    callable_name: &str,
+    expect: &Expect,
+) {
+    let callable_id = fir_package_store
+        .find_callable_id_by_name(callable_name)
+        .expect("callable should exist");
+
+    let callable_compute_properties = package_store_compute_properties.get_item(callable_id);
+    expect.assert_eq(&callable_compute_properties.to_string());
+}
+
+pub fn check_last_statement_compute_properties(
+    package_store_compute_properties: &PackageStoreComputeProperties,
+    expect: &Expect,
+) {
+    let last_package_id = package_store_compute_properties
+        .iter()
+        .map(|(package_id, _)| package_id)
+        .sorted()
+        .last()
+        .expect("at least one package should exist");
```

Comment:
This one is a tricky one... `sorted` seems like it it sorts on the fly, but internally it actually creates a copy of the vector, sorts it, and then returns a new owning iterator for the sorted items. In general, I would rather have the code do explicit copies and sort the copy so it's clear that copy is happening. That said, since all this code needs is the highest package id, which we can get from an stdlib function `max`:

```suggestion
    let last_package_id = package_store_compute_properties
        .iter()
        .map(|(package_id, _)| package_id)
        .max()
        .expect("at least one package should exist");
```